validphys2/src/validphys/commondataparser.py

```diff
@@ -761,13 +761,15 @@ def folder(self):
     def cm_energy(self):
         """Return the center of mass energy as GeV if it can be understood from the name
         otherwise return None"""
-        energy_string = self.setname.split("_")[2]
-        if energy_string == "NOTFIXED":
+
+        if "NOTFIXED" in self.setname:
```

Comment:
I think this is a bad idea. We need to change the `E<int>_NC` datasets to be `E<int>-NC`, otherwise we are already breaking the convention.

https://docs.nnpdf.science/data/dataset-naming-convention.html

Luckily we can use the same strategy that is used for the legacy versions to ensure old runcards still work.

Reply:
solved in https://github.com/NNPDF/nnpdf/pull/2142 and 697470ceb